### PR TITLE
Feature/ Add Reducers and Actions

### PR DESCRIPTION
**Pull request by Akorede3133**
In this PR, the folowing were effected;

* Created a directory that will contain all your Redux logic (/src/redux)
* Configured a Redux store (/src/redux/store.js)
* Defined a slice of state for books  and categories

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      }),
+    },
+    removeBook: (state, action) => {
+      const { id } = action.payload;
+      state.books.filter((book) => book.id !== id);
+    },
+  },
+});
+
+export default booksSlice.reducer;
```
Comment (TadesseDev) [inline src/redux/books/booksSlice.js]: - [ ] Dear @Akorede3133, Alongside the `booksSlice.reducer;` we need to export the `{ addBook, removeBook }` form `booksSlice.actions`. Please export these two actions.

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+};
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => state.categories.length && 'Under construction',
+  },
+});
+
+export default categoriesSlice.reducer;
```
Comment (TadesseDev) [inline src/redux/categories/categoriesSlice.js]: - [ ] Also here, Please export `{ checkStatus }` action from the `categoriesSlice.actions`,  we need this for later to dispatch an action.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Akorede3133 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by aamirkhan2478 — APPROVED** [review]
Hi @Akorede3133, 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly :+1: 
✅ No linter error :+1: 
✅ Good commit messages :+1: 
✅ Good descriptive PR :+1: 
✅ Well done :clap: you did all changes given by the previous reviewer :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**